dtfs-central-api/src/v1/validation/utilisation-report-service/selected-fee-record-validator.ts

```diff
@@ -10,3 +11,35 @@ export const validateSelectedFeeRecordsAllHaveSamePaymentCurrency = (selectedFee
     throw new InvalidPayloadError('Selected fee records must all have the same payment currency');
   }
 };
+
+/**
+ * Validates that all selected fee records with payments have the same fee
+ * record payment group and that the fee record IDs match the fee record IDs on
+ * this existing payment group.
+ * @param feeRecordIds - The fee record IDs
+ */
+export const validateSelectedFeeRecordsWithPaymentsAreOnePaymentGroup = async (feeRecordIds: number[]) => {
+  const selectedFeeRecords = await FeeRecordRepo.findByIdWithPaymentsAndFeeRecords(feeRecordIds);
+
+  const selectedFeeRecordStatuses = selectedFeeRecords.reduce((statuses, { status }) => statuses.add(status), new Set<FeeRecordStatus>());
+  if (selectedFeeRecordStatuses.size !== 1) {
+    throw new InvalidPayloadError('Fee records must all have the same status');
+  }
+
+  const firstFeeRecord = selectedFeeRecords[0];
+
+  if (firstFeeRecord.status !== FEE_RECORD_STATUS.DOES_NOT_MATCH) {
+    return;
+  }
+
+  const firstPayment = firstFeeRecord.payments[0];
+  // Sets would be faster for large volumes but not seen/needed here.
+  const savedFeeRecordIds = firstPayment.feeRecords.map((feeRecord) => feeRecord.id);
+
+  const isPaymentGroupExclusivelyConsistingOfIdsProvided =
+    feeRecordIds.every((id) => savedFeeRecordIds.includes(id)) && savedFeeRecordIds.length === feeRecordIds.length;
```

Comment:
I know I suggested this -- It might be best to do my first suggestion... If I sent [1,1,1] and the savedFeeRecordIds was [1,2,3] it would still pass... Not sure if that's an issue (we could also resolve with a `set`?